### ISSUE TITLE
[BUG]: Seg Fault when num_threads is bigger than number of cores

### ISSUE THREAD

**Issue by ncclementi** [nv-morpheus/Morpheus#2218]
### Version

25.06 nightly install

### Which installation method(s) does this occur on?

Conda

### Describe the bug.

I have a pipeline that is very similar to the [NLP sid example](https://github.com/nv-morpheus/Morpheus/tree/branch-25.06/examples/nlp_si_detection) but it uses the python API. You can see the code [here in this draft PR](https://github.com/rapidsai/deployment/pull/555/files/8139da22bf3a58a1b28746eed0263d5a3d4b5909#diff-8b3b46a3c27ebe005d882c88208a353ff1e59ff51a1ba7b8b787c035578ae1d9)

I had `config.num_threads = 16` to avoid default value of 1, and since I locally had 128 cores. See discussion https://github.com/nv-morpheus/Morpheus/issues/2214 for reference. 

Now when I moved from my local setup to a deployment in kubernetes, I triggered a segmentation fault. 

After we debugging this, we found out that the kubernetes node in this case had only 12 cores, and having `config.num_threads = 16` triggered a segmentation fault as soon as it tries to run the `PreprocessNLPStage(...)` it triggered the seg fault. 

I can confirm that as soon as `num_threads is > num_cores` = `len(os.sched_getaffinity(0))` in linux, the segfault happens. If it's below that, the pipeline runs successfully. 

**Suggestion for a fix:**

I'd expect that if someone tries to set `num_threads > num_cores` to Raise an Exception that uncovers the problem to the user instead of Seg faulting. 

cc: @jacobtomlinson  for visibility. 

### Minimum reproducible example

```shell
Run the SID NLP example with a num_threads bigger than the number of cores.
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

W20250515 22:20:37.678025 140424353273664 topology.cpp:143] user_cpuset was not a subset of topo_cpuset; 12 cpus were dropped
====Pipeline Pre-build====
====Pre-Building Segment: linear_segment_0====
====Pre-Building Segment Complete!====
====Pipeline Pre-build Complete!====
====Registering Pipeline====
====Building Pipeline====
====Building Pipeline Complete!====
F20250515 22:20:37.775679 140421606930112 edge_holder.hpp:61] A node was destructed which still had dependent connections. Nodes must be kept alive while dependent connections are still active
====Registering Pipeline Complete!====
====Starting Pipeline====
====Pipeline Started====
====Building Segment: linear_segment_0====
Added source: <from-kafka-0; KafkaSourceStage(bootstrap_servers=kafka-cluster-kafka-bootstrap:9092, input_topic=['network-traffic-input'], group_id=network-traffic-group, client_id=None, poll_interval=10millis, disable_commit=False, disable_pre_filtering=False, auto_offset_reset=earliest, stop_after=0, async_commits=True)>
  └─> morpheus.MessageMeta
Added stage: <deserialize-1; DeserializeStage(ensure_sliceable_index=True, task_type=None, task_payload=None)>
  └─ morpheus.MessageMeta -> morpheus.ControlMessage
Added stage: <message-filter-2; MessageFilterStage(c=Config(execution_mode=<ExecutionMode.GPU: 'GPU'>, debug=False, log_level=30, log_config_file=None, plugins=None, mode=<PipelineModes.NLP: 'NLP'>, _pipeline_batch_size=1024, _model_max_batch_size=32, feature_length=256, num_threads=16, edge_buffer_size=128, class_labels=['address', 'bank_acct', 'credit_card', 'email', 'govt_id', 'name', 'password', 'phone_num', 'secret_keys', 'user'], ae=None, fil=None, frozen=True), column=data, min_length=50)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <preprocess-nlp-3; PreprocessNLPStage(vocab_hash_file=data/bert-base-uncased-hash.txt, truncation=True, do_lower_case=True, add_special_tokens=False, stride=-1, column=data)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <inference-4; TritonInferenceStage(model_name=sid-minibert-onnx, server_url=tritonserver:8000, force_convert_inputs=True, use_shared_memory=False, needs_logits=None, inout_mapping=None, input_mapping=None, output_mapping=None, thread_count=None)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <monitor-5; MonitorStage(description=Inference Rate, smoothing=0.001, unit=inf, delayed_start=True, determine_count_fn=None, text_color=IndicatorsTextColor.cyan, font_style=IndicatorsFontStyle.bold, log_level=LogLevels.INFO)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <add-class-6; AddClassificationsStage(labels=None, prefix=, probs_type=TypeId.BOOL8, threshold=0.5)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <filter-7; FilterDetectionsStage(threshold=0.5, copy=True, filter_source=FilterSource.TENSOR, field_name=probs)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <network-traffic-analyzer-8; NetworkTrafficAnalyzerStage(c=Config(execution_mode=<ExecutionMode.GPU: 'GPU'>, debug=False, log_level=30, log_config_file=None, plugins=None, mode=<PipelineModes.NLP: 'NLP'>, _pipeline_batch_size=1024, _model_max_batch_size=32, feature_length=256, num_threads=16, edge_buffer_size=128, class_labels=['address', 'bank_acct', 'credit_card', 'email', 'govt_id', 'name', 'password', 'phone_num', 'secret_keys', 'user'], ae=None, fil=None, frozen=True))>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <serialize-9; SerializeStage(include=None, exclude=['^_ts_'], fixed_columns=True)>
  └─ morpheus.ControlMessage -> morpheus.MessageMeta
Added stage: <to-kafka-10; WriteToKafkaStage(bootstrap_servers=kafka-cluster-kafka-bootstrap:9092, output_topic=network-traffic-results, client_id=None)>
  └─ morpheus.MessageMeta -> morpheus.MessageMeta

</pre></details>

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

I created an environment with morpheus nightly, using this yaml

```yaml
channels:
  - conda-forge
  - nvidia/label/dev
  - rapidsai
dependencies:
  - python>=3.12
  - morpheus-core=25.06
```

</pre></details>

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by efajardo-nv** [comment]
Thanks @ncclementi. This will be addressed along with #2212 and #2214 to improve documentation on setting number of threads and more graceful/descriptive errors.
